typedload/dataloader.py

```diff
@@ -427,14 +427,28 @@ def _unionload(l: Loader, value, type_) -> Any:
     except AttributeError:
         raise TypedloadAttributeError('The typing API for this Python version is unknown')
 
+    value_type = type(value)
+
     # Do not convert basic types, if possible
-    if type(value) in args.intersection(l.basictypes):
+    if value_type in args.intersection(l.basictypes):
         return value
 
     exceptions = []
 
-    # Try all types
+    # Give a score to the types
+    sorted_args = []  # type: List[Type]
     for t in args:
+        if (value_type == list and is_list(t)) or \
+           (value_type == dict and is_dict(t)) or \
+           (value_type == frozenset and is_frozenset(t)) or \
+           (value_type == set and is_set(t)) or \
+           (value_type == tuple and is_tuple(t)):
```

Comment:
Would it possible to generalize this and use the type tests defined in the loader's handlers?

Reply:
I don't know, the issue is that `typing.List` and `list` are two completely separate things. And like this it works if the type is `List` or `List[t]` or `list`.